osu.Game/Overlays/SettingsPanel.cs

```diff
@@ -105,6 +108,13 @@ private void load()
             Add(SectionsContainer = new SettingsSectionsContainer
             {
                 Masking = true,
+                EdgeEffect = new EdgeEffectParameters
```

Comment:
Uh... I thought that was the point? Can you elaborate on (or better yet, point to) what's wrong on the screenshots, because I see nothing alarming?

Reply:
Maybe less of a shadow and more of a masking problem?

Before and After
![2RMxT143KG](https://github.com/ppy/osu/assets/35318437/ce63fc65-4796-47a4-aaa8-efa4a2da9517) ![PhzGlFtL41](https://github.com/ppy/osu/assets/35318437/9e96bf3c-0f88-4cea-8162-a05f456340d0)

Reply:
Ok so you *did* mean the left side.

Seems like applying the edge effect to `this` / the `SettingsPanel` itself rather than `SectionsContainer` would fix, but I'll wait and see if it was perhaps intentional, because it seems like applying to `SectionsContainer` would be the less obvious route.

Reply:
Probably fine? It's very subtle.

Reply:
It becomes more noticeable when the search text box section is pinned.

![osu!_OlZe3ZouyY](https://github.com/ppy/osu/assets/35318437/ccf8d7fc-d722-4533-9968-cad5cf1d9627)

And it depends on the window size, it can look noticeably worse:
![osu!_t7zjvQEJpE](https://github.com/ppy/osu/assets/35318437/0d54d364-2028-4ea4-90bd-99a7d818e29f)

Reply:
The issue with moving it a level up is it completely breaks nested panels (see `NonMaskedContent` usage).

I'll see if I can find another simple solution for this.

Reply:
It looks like a sub-pixel masking artifact as you mentioned. It *seems* like setting `MaskingSmoothness` fixes the issue for me, can you confirm?

```diff
diff --git a/osu.Game/Overlays/SettingsPanel.cs b/osu.Game/Overlays/SettingsPanel.cs
index 382423eb57..8a2bb3c22b 100644
--- a/osu.Game/Overlays/SettingsPanel.cs
+++ b/osu.Game/Overlays/SettingsPanel.cs
@@ -115,6 +115,7 @@ private void load()
                     Hollow = true,
                     Radius = 10
                 },
+                MaskingSmoothness = 0,
                 RelativeSizeAxes = Axes.Both,
                 ExpandableHeader = CreateHeader(),
                 SelectedSection = { BindTarget = CurrentSection },

```

Reply:
Yep, that fixes the artifacts.